File: analysis/import/SCMLogParser/src/test/java/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/NumberOfOccurencesInCommitsTest.java
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics;

import de.maibornwolff.codecharta.importer.scmlogparser.input.Modification;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class NumberOfOccurencesInCommitsTest {
    @Test
    public void should_have_initial_value_zero() {
        // when
        NumberOfOccurencesInCommits metric = new NumberOfOccurencesInCommits();

        // then
        assertThat(metric.value(metric.metricName())).isEqualTo(0L);
    }

    @Test
    public void should_increase_by_modification() {
        // given
        NumberOfOccurencesInCommits metric = new NumberOfOccurencesInCommits();

        // when
        metric.registerModification(new Modification("any"));

        // then
        assertThat(metric.value(metric.metricName())).isEqualTo(1L);
    }
}

File: analysis/import/SCMLogParser/src/test/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/NumberOfOccurencesInCommitsTest.kt
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics

import de.maibornwolff.codecharta.importer.scmlogparser.input.Modification
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class NumberOfOccurencesInCommitsTest {
    @Test
    fun should_have_initial_value_zero() {
        // when
        val metric = NumberOfOccurencesInCommits()
        // then
        assertThat(metric.value()).isEqualTo(0L)
    }

    @Test
    fun should_increase_by_modification() {
        // given
        val metric = NumberOfOccurencesInCommits()
        // when
        metric.registerModification(Modification("any"))
        // then
        assertThat(metric.value()).isEqualTo(1L)
    }
}